### PR TITLE
Filters Added for speaker

### PR DESCRIPTION
**Pull request by pAditya198**
## When None of the Filters are selected :

Screenshot (150)

## When Machine Learning is Selected as Filter :

Screenshot (151)

## When Mobile is Selected as Filter :

Screenshot (152)

### CODE REVIEW

**Comment by pAditya198** [comment]
Schedule Component Also Initialized
linked with speakers firebase

With dummy_data :

Screenshot (153)

With data :

Screenshot (154)

**Comment by pAditya198** [comment]
@samagragupta @aashutoshrathi Please review

Also Some UI fixes needed.

File: src/app/components/speakers-page/speakers-page.component.ts [hunk 1]
```diff
@@ -4,19 +4,110 @@ import { Speakers } from 'src/app/models/interfaces';
...
+            this.cloud_Active = true;
+        }
+    }
+    othersActive(value) {
+        this.othersCount += value;
+        if (this.othersCount % 2 == 0) {
+            this.others_Active = false;
+        } else {
+            this.others_Active = true;
+        }
```
Comment (aashutoshrathi) [inline src/app/components/speakers-page/speakers-page.component.ts]: You can also write the same things as:
```js
this.someActive = this.someCount%2;
```

Also, please use consistent styles, i.e. use only camelCase here.

File: src/app/components/speakers-page/speakers-page.component.ts [hunk 2]
```diff
@@ -4,19 +4,110 @@ import { Speakers } from 'src/app/models/interfaces';
...
+        this.web_Active = false;
+        this.mobile_Active = false;
+        this.ml_Active = false;
+        this.cloud_Active = false;
+        this.others_Active = false;
+        this.webCount = 0;
+        this.mobileCount = 0;
+        this.mlCount = 0;
+        this.cloudCount = 0;
+        this.othersCount = 0;
```
Comment (aashutoshrathi) [inline src/app/components/speakers-page/speakers-page.component.ts]: Use consistent cases here also.

File: src/app/components/speakers-page/speakers-page.component.ts [hunk 3]
```diff
@@ -4,19 +4,110 @@ import { Speakers } from 'src/app/models/interfaces';
...
+                        this.cloud.push(ele);
+                    }
+                    else if (ele.track == 'ml') {
+                        console.log(ele.track);
+                        this.ml.push(ele);
+                    } else {
+                        console.log(ele.track);
+                        this.others.push(ele);
+                    }
+                    return true
```
Comment (aashutoshrathi) [inline src/app/components/speakers-page/speakers-page.component.ts]: Remove all console logs from here

File: src/app/components/speakers-page/speakers-page.component.html [hunk 4]
```diff
@@ -9,18 +9,58 @@
...
+
     <div class="col-xs-12" *ngIf="!(speakers.enabled &&
             speakers.enabled.value)">
       <div class="sub-title" style='line-height:250px;font-size: 2.5em'>
         <p> Rockstars🤘 are on the way! </p>
       </div>
     </div>
-    <ng-container *ngIf="(speakers.enabled && speakers.enabled.value)">
+    <ng-container
+      *ngIf="(speakers.enabled && speakers.enabled.value && !web_Active && !mobile_Active && !ml_Active && !cloud_Active && !others_Active)">
```
Comment (aashutoshrathi) [inline src/app/components/speakers-page/speakers-page.component.html]: `speakers.enabled && speakers.enabled.value ` -> `speakers.enabled?.value `

**Comment by pAditya198** [comment]
Instead of button I've used Checkbox they look more classic and also less variables are used.
Some screenshot are provided below as an example of multiple checks.

Screenshot (156)


Screenshot (157)


Screenshot (158)


Screenshot (159)

**Comment by pAditya198** [comment]
Review changes are also fixed @aashutoshrathi

File: src/app/components/speakers-page/speakers-page.component.ts (hunk 1 above)
Comment (pAditya198) [reply src/app/components/speakers-page/speakers-page.component.ts]: instead I'll be using checkbox which will reduce some line of code and also will look much better

File: src/app/components/speakers-page/speakers-page.component.ts (hunk 2 above)
Comment (pAditya198) [reply src/app/components/speakers-page/speakers-page.component.ts]: okay

**Comment by pAditya198** [comment]
@aashutoshrathi should I create a new model for schedule because here I'm using the existing data from speakers

**Comment by AmanRaj1608** [comment]
@samagragupta @aashutoshrathi 
Is this schedule UI okay https://github.com/pAditya198/devfest-19/pull/7

**Comment by aashutoshrathi** [comment]
@samagragupta Check once more, we will merge it then

**Comment by pAditya198** [comment]
> @samagragupta @aashutoshrathi
> Is this schedule UI okay [pAditya198#7](https://github.com/pAditya198/devfest-19/pull/7)

@samagragupta @aashutoshrathi  suggest UI changes or should I just merge this commit.

**Comment by parth181195** [comment]
can nu finish the change @aashutoshrathi has requested so i can merge this in master

**Pull request merged**